flytekit/annotated/stuff.py

```diff
@@ -0,0 +1,451 @@
+            #       function.
+            bindings, upstream_nodes = self.interface.create_bindings_for_inputs(kwargs)
+
+            # Set upstream names. So for instance if we have
+            # (in a workflow...)
+            #     a = t1()  # This produces a promise.NodeOutput which contains a node
+            #     b = t2(a)  # When we're in b's __call__ we realize that a is an upstream node, so move up the stack
+            #                # to find the name 'a'
+            # Note that these upstream_nodes have been unpacked by the create_bindings_for_inputs function to be nodes
+            # and no longer promise.NodeOutputs
+            for n in upstream_nodes:
+                if n._id is None:
+                    logger.debug(f"Upstream node {n} is still missing a text id, moving up the call stack to find")
+                    n._id = get_earliest_promise_name(n)
+
+            # TODO: Make the metadata name the full name of the (function)?
+            # There should be no reason to ever assign a random node id (at least in a non-dynamic context), so we
+            # leave it empty for now.
+            sdk_node = _nodes.SdkNode(
+                id=None,
+                metadata=_workflow_model.NodeMetadata(self._task_function.__name__, self.metadata.timeout, self.metadata.retries,
+                                                      self.metadata.interruptible),
+                bindings=sorted(bindings, key=lambda b: b.var),
+                upstream_nodes=upstream_nodes,
+                sdk_task=self
+            )
+
+            # TODO: Return multiple versions of the _same_ node, but with different output names. This is what this
+            #       OutputParameterMapper does for us, but should we try to move away from it?
+            ppp = OutputParameterMapper(self.interface.outputs, sdk_node)
+            # Don't print this, it'll crash cuz sdk_node._upstream_node_ids might be None, but idl code will break
+
+            if len(self._outputs) > 1:
+                # Why do we need to do this? Just for proper binding downstream, nothing else. This is
+                # basically what happens when you call my_task_node.outputs.foo, but we're doing it for the user.
+                wrapped_nodes = [ppp[self._outputs[i]] for i in range(0, len(self._outputs))]
+                return tuple(wrapped_nodes)
+            else:
+                return ppp[self._outputs[0]]
+
+        else:
+            return self._task_function(*args, **kwargs)
+
+    @property
+    def interface(self) -> interface.TypedInterface:
+        return self._interface
+
+    @property
+    def metadata(self) -> _task_model.TaskMetadata:
+        return self._metadata
+
+
+def task(
+        _task_function=None,
+        outputs: List[str]=None,
+        cache_version='',
+        retries=0,
+        interruptible=None,
+        deprecated='',
+        storage_request=None,
```

Comment:
i want to club the requests and limits as structs. This simplifies adding new ones